csharp/src/Ice/SslEngine.cs

```diff
@@ -3,467 +3,232 @@
-    /// the private key associated with the certificate in the given file.</summary>
-    /// <param name="file">The certificate file name.</param>
-    /// <returns>The password for the key or null, if no password is necessary.</returns>
-    public delegate SecureString IPasswordCallback(string file);
-
     internal class SslEngine
     {
-        internal X509Certificate2Collection? CaCerts { get; }
-        internal X509Certificate2Collection? Certs { get; }
-        internal LocalCertificateSelectionCallback? CertificateSelectionCallback { get; }
-        internal RemoteCertificateValidationCallback? RemoteCertificateValidationCallback { get; }
-        internal int CheckCRL { get; }
-        internal IPasswordCallback? PasswordCallback { get; }
+        // Specifies the mode used to check for X509 certificate revocation when validating client certificates.
+        internal X509RevocationMode ClientCertificateRevocationCheckMode { get; }
+
+        // A collection of X509 certificates to use by outgoing connections
+        internal X509Certificate2Collection? ClientCertificates;
+
+        // A certificate selection callback used to select the client side certificate.
+        internal LocalCertificateSelectionCallback? ClientCertificateSelectionCallback { get; }
+
+        // The list of SSL protocols to enable for outgoing connections.
+        internal SslProtocols ClientEnabledSslProtocols { get; }
+
+        // The certificates collection that will be used as trusted certificate authorities to verify the client
+        // certificate.
+        internal X509Certificate2Collection? ClientCertificateCertificateAuthorities { get; }
+
+        // The callback that will be used to verify the client certificates.
+        internal RemoteCertificateValidationCallback? ClientCertificateValidationCallback { get; }
+
+        // A boolean value indicating whenever or not a client certificate is required by the server
+        internal bool RequireClientCertificate { get; }
+
+        // Specifies the mode used to check for X509 certificate revocation when validating server certificates.
+        internal X509RevocationMode ServerCertificateRevocationCheckMode { get; }
+
+        // An X509 certificate to use by incoming connections
+        internal X509Certificate2? ServerCertificate;
+
+        // The list of SSL protocols to enable for incoming connections.
+        internal SslProtocols ServerEnabledSslProtocols { get; }
+
+        // The certificates collection that will be used as trusted certificate authorities to verify the server
+        // certificate.
+        internal X509Certificate2Collection? ServerCertificateCertificateAuthorities { get; }
+
+        // The callback that will be used to verify the server certificates.
+        internal RemoteCertificateValidationCallback? ServerCertificateValidationCallback { get; }
```

Comment:
I do it this way because the tls option classes have red/write properties, and we don't want the changes after it was first created. If/when we move to C# 9 we can use init only properties and data records. I guess we can keep a copy of the classes so users cannot modify it after initialization.